packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -514,6 +575,54 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		cursor.free();
 		return trees;
 	}
+
+	private purgeRevertibles(): void {
+		for (const revertible of this.revertibles) {
+			revertible.dispose();
+		}
+	}
+
+	private disposeRevertible(revertible: DisposableRevertible, revision: RevisionTag): void {
+		this.revertibleCommitBranches.delete(revision);
```

Comment:
does the branch also need to be explicitly disposed ?

Reply:
It sure does, thank you for catching that!